src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -816,6 +821,46 @@ private DeleteOrganisationResponse deleteOrganisationEntity(Organisation organis
         return deleteOrganisationResponse;
     }
 
+    @Override
+    @Transactional
+    public DeleteUserResponse deleteUserForOrganisation(List<String> emails) {
+        var deleteOrganisationResponse = new DeleteOrganisationResponse();
+        if (emails.isEmpty()) {
+            throw new InvalidRequest("Please provide both email addresses");
+        }
+
+        Set<String> userIdsToBeDeleted = emails.stream().map(
+            email -> {
+                Set<String> userIds = new HashSet<>();
+                Optional.ofNullable(professionalUserRepository
+                        .findByEmailAddress(RefDataUtil.removeAllSpaces(email)))
+                    .ifPresentOrElse(professionalUser -> {
+                        userIds.add(professionalUser.getUserIdentifier());
+                        userAttributeRepository.deleteByProfessionalUserId(professionalUser.getId());
+                        professionalUserRepository.delete(professionalUser);
+                    }, () -> {
+                            throw new InvalidRequest("Email addresses provided do not exist"); });
+                return userIds.stream().iterator().next();
+            }).collect(Collectors.toSet());
+
+        if (userIdsToBeDeleted.isEmpty()) {
+            throw new InvalidRequest("Could not find users to delete");
+        }
+
+        DeleteUserProfilesRequest deleteUserProfileRequest =
+            new DeleteUserProfilesRequest(userIdsToBeDeleted.stream().collect(
+            Collectors.toSet()));
+        deleteOrganisationResponse = RefDataUtil
+            .deleteUserProfilesFromUp(deleteUserProfileRequest, userProfileFeignClient);
+        if (deleteOrganisationResponse == null) {
```

Comment:
this line is not required, response will contain either success or error status or it wont come back if any FeignException